docs/gcp_kms.md

```diff
@@ -0,0 +1,190 @@
+---
+id: gcp_kms
+title: CloudKMS
+---
+
+##
+**Google Cloud Platform configuration**
+
+Create a new project or use an existing project and the service accounts used with Signatory should have the following permissions. It may be achieved by using custom roles (see [https://console.cloud.google.com/iam-admin/roles](https://console.cloud.google.com/iam-admin/roles)) \
+Project name is required in the signatory config.
+
+###
+**Basic permissions**
+* `cloudkms.cryptoKeyVersions.get`
+* `cloudkms.cryptoKeyVersions.list`
+* `cloudkms.cryptoKeyVersions.viewPublicKey`
+* `cloudkms.cryptoKeys.get`
+* `cloudkms.cryptoKeys.list`
+
+###
+**Sign**
+* `cloudkms.cryptoKeyVersions.useToSign`
+###
+**Import**
+
+* `cloudkms.cryptoKeyVersions.create`
+* `cloudkms.cryptoKeys.create`
+* `cloudkms.importJobs.create`
+* `cloudkms.importJobs.get`
+* `cloudkms.importJobs.list`
+* `cloudkms.importJobs.useToImport`
+
+###
+**Configuration parameters \
+Below are the configuration fields which are required for Signatory.**
+
+<table>
+  <tr>
+   <td>
+<strong>Name</strong>
+   </td>
+   <td><strong>Type</strong>
+   </td>
+   <td><strong>Required</strong>
+   </td>
+   <td><strong>Description</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>application_credentials
+   </td>
+   <td>string
+   </td>
+   <td>
+   </td>
+   <td>Path to the GCP application token JSON file (overrides <code>GOOGLE_APPLICATION_CREDENTIALS</code> environment variable)
+   </td>
+  </tr>
+  <tr>
+   <td>application_credentials_data
+   </td>
+   <td>string
+   </td>
+   <td>
+   </td>
+   <td>GCP application token JSON data (overrides <code>application_credentials</code>)
+   </td>
+  </tr>
+  <tr>
+   <td>project
+   </td>
+   <td>string
+   </td>
+   <td>✅
```

Comment:
Similar to how there is a ✅ do you want to use something like ❌ for not required fields? Right now it's just empty and I think it looks better to have something in there as opposed to leaving it empty

Reply:
Instead of the cross symbol shall I just add the word "OPTIONAL" itself? Because X feels like the field is deprecated or something which the user should not use or cannot use.

Reply:
@AbineshECAD yeah that sounds good to me